arm-network/2016-12-01/swagger/applicationGateway.json

```diff
@@ -0,0 +1,1331 @@
+        }
+      },
+      "description": "Properties of Frontend IP configuration of an application gateway."
+    },
+    "ApplicationGatewayFrontendIPConfiguration": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationGatewayFrontendIPConfigurationPropertiesFormat"
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the resource that is unique within a resource group. This name can be used to access the resource."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "Frontend IP configuration of an application gateway."
+    },
+    "ApplicationGatewayFrontendPortPropertiesFormat": {
+      "properties": {
+        "port": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Frontend port"
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "Provisioning state of the frontend port resource. Possible values are: 'Updating', 'Deleting', and 'Failed'."
+        }
+      },
+      "description": "Properties of Frontend port of an application gateway."
+    },
+    "ApplicationGatewayFrontendPort": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationGatewayFrontendPortPropertiesFormat"
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the resource that is unique within a resource group. This name can be used to access the resource."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "Frontend port of an application gateway."
+    },
+    "ApplicationGatewayBackendAddress": {
+      "properties": {
+        "fqdn": {
+          "type": "string",
+          "description": "Fully qualified domain name (FQDN)."
+        },
+        "ipAddress": {
+          "type": "string",
+          "description": "IP address"
+        }
+      },
+      "description": "Backend address of an application gateway."
+    },
+    "ApplicationGatewayBackendAddressPoolPropertiesFormat": {
+      "properties": {
+        "backendIPConfigurations": {
+          "type": "array",
+          "items": {
+            "$ref": "./../../2016-09-01/swagger/networkInterface.json#/definitions/NetworkInterfaceIPConfiguration"
```

Comment:
@jobatzil  This model has a property named "properties" of type ["NetworkInterfaceIPConfigurationPropertiesFormat"](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-network/2016-09-01/swagger/networkInterface.json#L335), which contains a property named "applicationGatewayBackendAddressPools" that references "./applicationGateway.json#/definitions/ApplicationGatewayBackendAddressPool".

The above reference is to the 2016-09-01 version of applicationgateway.json.

Isn't that weird. Shouldn't it be referencing the 2016-12-01 version of applicationgateway.json (I mean circling back to this file)?

/cc @deepakrajendranmsft, @johanste, @olydis